catroid/src/test/java/org/catrobat/catroid/test/ui/recyclerview/UserDataDeletionTest.java

```diff
@@ -0,0 +1,328 @@
+
+		setUpScene1();
+		setUpScene2();
+
+		ProjectManager.getInstance().setCurrentProject(project);
+	}
+
+	private void setUpScene1() {
+		Script script = new StartScript();
+		scene1 = new Scene();
+		sprite1 = new Sprite();
+
+		sprite1.addUserList(localListInUse);
+		sprite1.addUserList(localListNotInUse);
+		sprite1.addUserVariable(localVariableInUse);
+		sprite1.addUserVariable(localVariableNotInUse);
+		sprite1.addUserVariable(localVariableInUseDropDown);
+
+		FormulaElement formulaElementSetTransparency =
+				new FormulaElement(FormulaElement.ElementType.USER_LIST, localListInUse.getName(), null);
+		Formula formulaSetTransparency = new Formula(formulaElementSetTransparency);
+		FormulaBrick setTransparency = new ChangeXByNBrick(formulaSetTransparency);
+		script.addBrick(setTransparency);
+
+		FormulaElement formulaElementChangeXByN = new FormulaElement(FormulaElement.ElementType.USER_VARIABLE,
+				localVariableInUse.getName(), null);
+		Formula formulaChangeXByN = new Formula(formulaElementChangeXByN);
+		FormulaBrick changeXByNBrick = new ChangeXByNBrick(formulaChangeXByN);
+		script.addBrick(changeXByNBrick);
+
+		FormulaElement formulaElementChangeVariable =
+				new FormulaElement(FormulaElement.ElementType.NUMBER, STRING_ONE, null);
+		Formula formulaChangeVariable = new Formula(formulaElementChangeVariable);
+		changeVariableBrick = new ChangeVariableBrick(formulaChangeVariable, localVariableInUseDropDown);
+		script.addBrick(changeVariableBrick);
+
+		sprite1.addScript(script);
+		scene1.addSprite(sprite1);
+		project.addScene(scene1);
+	}
+
+	private void setUpScene2() {
+		Script script = new StartScript();
+		scene2 = new Scene();
+		sprite2 = new Sprite();
+
+		sprite2.addUserList(new UserList("LocalListNotInUse"));
+
+		FormulaElement formulaElementSetTransparency = new FormulaElement(FormulaElement.ElementType.USER_VARIABLE,
+				globalVariableInUse.getName(), null);
+		Formula formulaSetTransparency = new Formula(formulaElementSetTransparency);
+		FormulaBrick setTransparency = new ChangeXByNBrick(formulaSetTransparency);
+		script.addBrick(setTransparency);
+
+		sprite2.addScript(script);
+		scene2.addSprite(sprite2);
+		project.addScene(scene2);
+	}
+
+	private void setUpProjectManager(Scene scene, Sprite sprite) {
+		ProjectManager.getInstance().setCurrentlyEditedScene(scene);
+		ProjectManager.getInstance().setCurrentSprite(sprite);
+		activityController.create().resume();
+		spyDataListFragment.initializeAdapter();
+	}
+
+	private List<Brick> getBrickList(List<Scene> sceneList, @Nullable Sprite currentSprite) {
+
+		List<Brick> brickFlatList = new ArrayList();
+
+		for (Scene scene : sceneList) {
+			for (Sprite sprite : scene.getSpriteList()) {
+				if (sprite == currentSprite || currentSprite == null) {
+					for (Script script : sprite.getScriptList()) {
+						script.addToFlatList(brickFlatList);
+					}
+				}
+			}
+		}
+
+		return brickFlatList;
+	}
+
+	@Test
+	public void testDelete() {
```

Comment:
To reference a good clean code test, each test function should only test one specific function, which can often result in one Assert for one Test method. The whole testclass consists of a lot of complex setup and nestings already in the setup so its quite hard to read whats going on here. If possible make the setup much more easier and add your specific setups to one test method. Best case this should result in more Test methods with about 5 to 10 lines of code and an Assert at the end. 

Testing everything within just one function will also be not very helpful in terms of Continuous Integration since this test wont tell you where's the problem at. e.g. Jenkins runs another PR and it tells you "testDelete" failed. You now have to debug your test first to find out where the problem occurs